mne/beamformer/_compute_beamformer.py

```diff
@@ -261,8 +263,12 @@ def _compute_beamformer(G, Cm, reg, n_orient, weight_norm, pick_ori,
                 # Free source orientation
                 if inversion == 'single':
                     # Invert for each dipole separately using plain division
+                    diags = np.diagonal(Ck, axis1=1, axis2=2)
+                    if reduce_rank:
+                        diags[np.arange(len(diags)),
+                              np.argmin(diags, axis=1)] = np.inf
```

Comment:
Why would you ever want to reduce rank when `inversion=='single'`? The whole point of this mode is to avoid matrix inversion issues.

Reply:
In theory you could have one of those three directions nulled in the lead field -- sources at locations that have x=y=0 or x=z=0, or y=z=0 in the MEG case will have a zero lead field in the z, y, or x direction (respectively) in the spherical conductor case. I'm assuming that `'single'` will not save you in this case, but maybe it does? Or maybe I'm thinking about it incorrectly? If I'm thinking about it correctly, then maybe `'single'` shouldn't even be used if there is a spherical conductor...?

In any case, what should it do in the `single` case, raise an error if `reduce_rank=True` and `inversion='single'`? That seems like the most expedient choice, as we could think about whether or not something smarter can be done later.

Reply:
Ah I understand now. In the conpy paper, we used a forward modal with only 2 dipoles at each source point (a "tangential" leadfield) for just this reason: https://github.com/AaltoImagingLanguage/conpy/blob/master/conpy/forward.py#L455

![forward2](https://user-images.githubusercontent.com/428273/71986959-01eaa880-3236-11ea-809b-a55e24bdfd05.png)

In the `inversion='matrix'` case, I guess zeroing out the smallest eigenvalue will do something very similar: eliminate the radial component.

However, in the above `inversion='single'` case, the XYZ dipoles are pointing relative to the cortical surface, not relative to the radial component. So in all likelihood, you are zapping the wrong direction. I propose the following: do an eigen-decomposition of all three dipoles, then take the two largest eigenvalues to be two new dipoles. Then proceed to invert those two dipoles individually.

Reply:
> I propose the following: do an eigen-decomposition of all three dipoles, then take the two largest eigenvalues/vectors to be two new dipoles. Then proceed to invert those two dipoles individually.

This should end up being equivalent to the `'matrix'` inversion mode, right? If so then I propose instead to just raise an error when `'single'` is used with `reduce_rank=True` and tell people to use the `'matrix'` mode.

Reply:
> This should end up being equivalent to the `'matrix'` inversion mode, right?

Would it? In that case, yeah, let's just throw an error.

Reply:
`reduce_rank=True` should in practice never be needed when `inversion='single'` anyway.